### DISCUSSION TITLE
基座安装的UI组件共享给子应用

### DISCUSSION THREAD

**Discussion by sitonlotus** [jd-opensource/micro-app#735] [General]
你好，我想要的是在基座中通过npm/yarn/cnpm等包管理工具安装UI组件，然后引入后，子应用能够直接使用，这样在打包的时候就可以资源，在加载的时候可以减少文件资源的重复加载，这样还顺畅，这样能实现吗？
![image](https://user-images.githubusercontent.com/41383577/226096477-48ce8d3b-088b-4545-ad4d-cefd3999de87.png)

**Comment by bailicangdu** [comment]
这里的共享js只是子应用之间的贡献，基座共享组件库是很难做到的，尤其是按需加载的组件

**Comment by X-W-W** [comment]
可以使用模块联邦进行共享操作

**Reply by uioonm** [reply]
请问大佬有接入模块联邦的demo吗，我在接入的时候遇到Uncaught (in promise) ScriptExternalLoadError: Loading script failed.的报错，看了历史的issue貌似没有解决方案，请问你是怎么解决这个问题的呢

**Reply by X-W-W** [reply]
我这边是在主应用上使用@originjs/vite-plugin-federation将组件分享出去
